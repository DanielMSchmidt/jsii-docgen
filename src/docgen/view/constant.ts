import * as reflect from 'jsii-reflect';
import { Markdown } from '../render/markdown';
import { PropertySchema } from '../schema';
import { Transpile, TranspiledType } from '../transpile/transpile';
import { Property } from './property';

export class Constant {
  private readonly constant: Property;
  constructor(transpile: Transpile, property: reflect.Property, linkFormatter: (type: TranspiledType) => string) {
    this.constant = new Property(transpile, property, linkFormatter);
  }
  public toMarkdown(): Markdown {
    return this.constant.toMarkdown();
  }

  public toJson(): PropertySchema {
    return this.constant.toJson();
  }
}
