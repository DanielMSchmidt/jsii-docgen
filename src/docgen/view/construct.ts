import * as reflect from 'jsii-reflect';
import { Markdown } from '../render/markdown';
import { ConstructSchema } from '../schema';
import { Transpile, TranspiledType } from '../transpile/transpile';
import { Class } from './class';

export class Construct {
  private readonly construct: Class;
  constructor(transpile: Transpile, klass: reflect.ClassType, linkFormatter: (type: TranspiledType) => string) {
    this.construct = new Class(transpile, klass, linkFormatter);
  }
  public toMarkdown(): Markdown {
    return this.construct.toMarkdown();
  }

  public toJson(): ConstructSchema {
    return this.construct.toJson();
  }
}
