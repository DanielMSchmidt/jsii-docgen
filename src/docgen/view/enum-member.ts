import * as reflect from 'jsii-reflect';
import { Markdown } from '../render/markdown';
import { EnumMemberSchema } from '../schema';
import { Transpile, TranspiledEnumMember } from '../transpile/transpile';

export class EnumMember {
  private readonly transpiled: TranspiledEnumMember;
  constructor(transpile: Transpile, private readonly em: reflect.EnumMember) {
    this.transpiled = transpile.enumMember(em);
  }
  public toMarkdown(): Markdown {
    const md = new Markdown({
      id: `${this.transpiled.fqn}`,
      header: {
        title: this.transpiled.name,
        pre: true,
        strike: this.em.docs.deprecated,
      },
    });

    if (this.em.docs.deprecated) {
      md.bullet(
        `${Markdown.italic('Deprecated:')} ${this.em.docs.deprecationReason}`,
      );
      md.lines('');
    }

    if (this.em.docs) {
      md.docs(this.em.docs);
    }

    md.split();
    md.lines('');

    return md;
  }

  public toJson(): EnumMemberSchema {
    return {
      id: this.transpiled.fqn,
      name: this.transpiled.name,
      deprecated: this.em.docs.deprecated,
      deprecationReason: this.em.docs.deprecationReason,
      docs: this.em.docs.toString(),
    };
  }
}
