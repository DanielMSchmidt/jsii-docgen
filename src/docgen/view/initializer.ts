import * as reflect from 'jsii-reflect';
import { Markdown } from '../render/markdown';
import { InitializerSchema } from '../schema';
import { Transpile, TranspiledCallable, TranspiledType } from '../transpile/transpile';
import { Parameter } from './parameter';

export class Initializer {
  private readonly transpiled: TranspiledCallable;
  private readonly parameters: Parameter[];
  constructor(
    private readonly transpile: Transpile,
    initializer: reflect.Initializer,
  ) {
    this.transpiled = transpile.callable(initializer);
    this.parameters = this.transpiled.parameters.map(
      (p) => new Parameter(this.transpile, p),
    );
  }

  public toMarkdown(
    linkFormatter: (type: TranspiledType) => string,
  ): Markdown {
    const md = new Markdown({
      id: `${this.transpiled.parentType.fqn}.Initializer`,
      header: {
        title: 'Initializers',
      },
    });

    md.code(
      this.transpile.language.toString(),
      `${this.transpiled.import}`,
      '',
      ...this.transpiled.invocations,
    );

    for (const parameter of this.parameters) {
      md.section(parameter.toMarkdown(linkFormatter));
    }

    return md;
  }

  public toJson(): InitializerSchema {
    const md = new Markdown();
    md.code(
      this.transpile.language.toString(),
      `${this.transpiled.import}`,
      '',
      ...this.transpiled.invocations,
    );

    return {
      id: `${this.transpiled.parentType.fqn}.Initializer`,
      snippet: md.render(),
      parameters: this.parameters.map((parameter) => parameter.toJson()),
    };
  }
}
