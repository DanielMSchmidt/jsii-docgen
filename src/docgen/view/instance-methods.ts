import * as reflect from 'jsii-reflect';
import { Markdown } from '../render/markdown';
import { MethodSchema } from '../schema';
import { Transpile, TranspiledType } from '../transpile/transpile';
import { InstanceMethod } from './instance-method';

export class InstanceMethods {
  private readonly instanceMethods: InstanceMethod[];
  constructor(transpile: Transpile, methods: reflect.Method[]) {
    this.instanceMethods = methods
      .filter((m) => !m.protected && !m.static)
      .map((m) => new InstanceMethod(transpile, m));
  }

  public toMarkdown(linkFormatter: (type: TranspiledType) => string): Markdown {
    if (this.instanceMethods.length === 0) {
      return Markdown.EMPTY;
    }

    const md = new Markdown({ header: { title: 'Methods' } });
    for (const method of this.instanceMethods) {
      md.section(method.toMarkdown(linkFormatter));
    }
    return md;
  }

  public toJson(): MethodSchema[] {
    return this.instanceMethods.map((method) => method.toJson());
  }
}
