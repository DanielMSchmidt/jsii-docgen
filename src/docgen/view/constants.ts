import * as reflect from 'jsii-reflect';
import { Markdown } from '../render/markdown';
import { PropertySchema } from '../schema';
import { Transpile, TranspiledType } from '../transpile/transpile';
import { Constant } from './constant';

export class Constants {
  private readonly constants: Constant[];
  constructor(transpile: Transpile, properties: reflect.Property[]) {
    this.constants = properties
      .filter((p) => !p.protected && p.const)
      .map((p) => new Constant(transpile, p));
  }

  public toMarkdown(linkFormatter: (type: TranspiledType) => string): Markdown {
    if (this.constants.length === 0) {
      return Markdown.EMPTY;
    }

    const md = new Markdown({ header: { title: 'Constants' } });
    for (const c of this.constants) {
      md.section(c.toMarkdown(linkFormatter));
    }
    return md;
  }

  public toJson(): PropertySchema[] {
    return this.constants.map((constant) => constant.toJson());
  }
}
