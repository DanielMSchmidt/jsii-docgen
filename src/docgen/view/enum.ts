import * as reflect from 'jsii-reflect';
import { Markdown } from '../render/markdown';
import { EnumSchema } from '../schema';
import { Transpile } from '../transpile/transpile';
import { EnumMember } from './enum-member';

export class Enum {
  private readonly members: EnumMember[];
  constructor(
    private readonly transpile: Transpile,
    private readonly enu: reflect.EnumType,
  ) {
    this.members = enu.members.map((em) => new EnumMember(transpile, em));
  }

  public toMarkdown(): Markdown {
    const transpiled = this.transpile.enum(this.enu);
    const md = new Markdown({ header: { title: transpiled.name } });

    if (this.enu.docs) {
      md.docs(this.enu.docs);
    }

    for (const m of this.members) {
      md.section(m.toMarkdown());
    }

    return md;
  }

  public toJson(): EnumSchema {
    const transpiled = this.transpile.enum(this.enu);
    return {
      name: transpiled.name,
      docs: this.enu.docs.toString(),
      members: this.members.map((member) => member.toJson()),
    };
  }
}
