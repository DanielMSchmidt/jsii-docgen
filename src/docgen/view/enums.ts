import * as reflect from 'jsii-reflect';
import { Markdown } from '../render/markdown';
import { EnumSchema } from '../schema';
import { Transpile } from '../transpile/transpile';
import { Enum } from './enum';

export class Enums {
  private readonly enums: Enum[];
  constructor(transpile: Transpile, enums: reflect.EnumType[]) {
    this.enums = enums.map((e) => new Enum(transpile, e));
  }

  public toMarkdown(): Markdown {
    if (this.enums.length === 0) {
      return Markdown.EMPTY;
    }

    const md = new Markdown({ header: { title: 'Enums' } });
    for (const e of this.enums) {
      md.section(e.toMarkdown());
    }
    return md;
  }

  public toJson(): EnumSchema[] {
    return this.enums.map((e) => e.toJson());
  }
}
