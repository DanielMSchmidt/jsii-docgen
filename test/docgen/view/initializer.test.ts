import * as reflect from 'jsii-reflect';
import { CSharpTranspile } from '../../../src/docgen/transpile/csharp';
import { JavaTranspile } from '../../../src/docgen/transpile/java';
import { PythonTranspile } from '../../../src/docgen/transpile/python';
import { TranspiledType } from '../../../src/docgen/transpile/transpile';
import { TypeScriptTranspile } from '../../../src/docgen/transpile/typescript';
import { Initializer } from '../../../src/docgen/view/initializer';
import { Assemblies } from '../assemblies';

const assembly: reflect.Assembly = Assemblies.instance.withoutSubmodules;

const findInitializer = (): reflect.Initializer => {
  for (const klass of assembly.system.classes) {
    if (klass.initializer) {
      return klass.initializer;
    }
  }
  throw new Error('Assembly does not contain an initializer');
};

describe('python', () => {
  const transpile = new PythonTranspile();
  test('markdown snapshot', () => {
    const initializer = new Initializer(transpile, findInitializer());
    expect(initializer.toMarkdown((t: TranspiledType) => `#${t.fqn}`).render()).toMatchSnapshot();
  });

  test('json snapshot', () => {
    const initializer = new Initializer(transpile, findInitializer());
    expect(initializer.toJson()).toMatchSnapshot();
  });
});

describe('typescript', () => {
  const transpile = new TypeScriptTranspile();
  test('markdown snapshot', () => {
    const initializer = new Initializer(transpile, findInitializer());
    expect(initializer.toMarkdown((t: TranspiledType) => `#${t.fqn}`).render()).toMatchSnapshot();
  });

  test('json snapshot', () => {
    const initializer = new Initializer(transpile, findInitializer());
    expect(initializer.toJson()).toMatchSnapshot();
  });
});

describe('java', () => {
  const transpile = new JavaTranspile();
  test('markdown snapshot', () => {
    const initializer = new Initializer(transpile, findInitializer());
    expect(initializer.toMarkdown((t: TranspiledType) => `#${t.fqn}`).render()).toMatchSnapshot();
  });

  test('json snapshot', () => {
    const initializer = new Initializer(transpile, findInitializer());
    expect(initializer.toJson()).toMatchSnapshot();
  });
});

describe('csharp', () => {
  const transpile = new CSharpTranspile();
  test('markdown snapshot', () => {
    const initializer = new Initializer(transpile, findInitializer());
    expect(initializer.toMarkdown((t: TranspiledType) => `#${t.fqn}`).render()).toMatchSnapshot();
  });

  test('json snapshot', () => {
    const initializer = new Initializer(transpile, findInitializer());
    expect(initializer.toJson()).toMatchSnapshot();
  });
});
