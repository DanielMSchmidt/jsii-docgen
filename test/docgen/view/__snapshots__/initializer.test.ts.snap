// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csharp json snapshot 1`] = `
Object {
  "id": "Amazon.CDK.AWS.ECR.CfnPublicRepository.Initializer",
  "parameters": Array [
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "- scope in which this resource is defined.",
      "id": "Amazon.CDK.AWS.ECR.CfnPublicRepository.Scope",
      "name": "Scope",
      "optional": false,
      "type": Object {
        "fqn": "@aws-cdk/core.Construct",
        "name": "Amazon.CDK.Construct",
      },
    },
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "- scoped id of the resource.",
      "id": "Amazon.CDK.AWS.ECR.CfnPublicRepository.Id",
      "name": "Id",
      "optional": false,
      "type": Object {
        "name": "string",
      },
    },
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "- resource properties.",
      "id": "Amazon.CDK.AWS.ECR.CfnPublicRepository.Props",
      "name": "Props",
      "optional": true,
      "type": Object {
        "fqn": "@aws-cdk/aws-ecr.CfnPublicRepositoryProps",
        "name": "Amazon.CDK.AWS.ECR.CfnPublicRepositoryProps",
      },
    },
  ],
  "snippet": "\`\`\`csharp
using Amazon.CDK.AWS.ECR;

new CfnPublicRepository(Construct Scope, string Id, CfnPublicRepositoryProps Props = null);
\`\`\`
",
}
`;

exports[`csharp markdown snapshot 1`] = `
" Initializers <a name=\\"Amazon.CDK.AWS.ECR.CfnPublicRepository.Initializer\\"></a>

\`\`\`csharp
using Amazon.CDK.AWS.ECR;

new CfnPublicRepository(Construct Scope, string Id, CfnPublicRepositoryProps Props = null);
\`\`\`

# \`Scope\`<sup>Required</sup> <a name=\\"Amazon.CDK.AWS.ECR.CfnPublicRepository.parameter.Scope\\"></a>

- *Type:* [\`Amazon.CDK.Construct\`](#Amazon.CDK.Construct)

scope in which this resource is defined.

---

# \`Id\`<sup>Required</sup> <a name=\\"Amazon.CDK.AWS.ECR.CfnPublicRepository.parameter.Id\\"></a>

- *Type:* \`string\`

scoped id of the resource.

---

# \`Props\`<sup>Optional</sup> <a name=\\"Amazon.CDK.AWS.ECR.CfnPublicRepository.parameter.Props\\"></a>

- *Type:* [\`Amazon.CDK.AWS.ECR.CfnPublicRepositoryProps\`](#Amazon.CDK.AWS.ECR.CfnPublicRepositoryProps)

resource properties.

---
"
`;

exports[`java json snapshot 1`] = `
Object {
  "id": "software.amazon.awscdk.services.ecr.CfnPublicRepository.Initializer",
  "parameters": Array [
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "- scope in which this resource is defined.",
      "id": "software.amazon.awscdk.services.ecr.CfnPublicRepository.scope",
      "name": "scope",
      "optional": false,
      "type": Object {
        "fqn": "@aws-cdk/core.Construct",
        "name": "software.amazon.awscdk.core.Construct",
      },
    },
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "- scoped id of the resource.",
      "id": "software.amazon.awscdk.services.ecr.CfnPublicRepository.id",
      "name": "id",
      "optional": false,
      "type": Object {
        "name": "java.lang.String",
      },
    },
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "\`AWS::ECR::PublicRepository.RepositoryCatalogData\`.",
      "id": "software.amazon.awscdk.services.ecr.CfnPublicRepositoryProps.repositoryCatalogData",
      "name": "repositoryCatalogData",
      "optional": true,
      "type": Object {
        "name": "java.lang.Object",
      },
    },
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "\`AWS::ECR::PublicRepository.RepositoryName\`.",
      "id": "software.amazon.awscdk.services.ecr.CfnPublicRepositoryProps.repositoryName",
      "name": "repositoryName",
      "optional": true,
      "type": Object {
        "name": "java.lang.String",
      },
    },
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "\`AWS::ECR::PublicRepository.RepositoryPolicyText\`.",
      "id": "software.amazon.awscdk.services.ecr.CfnPublicRepositoryProps.repositoryPolicyText",
      "name": "repositoryPolicyText",
      "optional": true,
      "type": Object {
        "name": "java.lang.Object",
      },
    },
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "\`AWS::ECR::PublicRepository.Tags\`.",
      "id": "software.amazon.awscdk.services.ecr.CfnPublicRepositoryProps.tags",
      "name": "tags",
      "optional": true,
      "type": Object {
        "name": "java.util.List<%>",
        "types": Array [
          Object {
            "fqn": "@aws-cdk/core.CfnTag",
            "name": "software.amazon.awscdk.core.CfnTag",
          },
        ],
      },
    },
  ],
  "snippet": "\`\`\`java
import software.amazon.awscdk.services.ecr.CfnPublicRepository;

CfnPublicRepository.Builder.create(Construct scope, java.lang.String id)
//  .repositoryCatalogData(java.lang.Object)
//  .repositoryName(java.lang.String)
//  .repositoryPolicyText(java.lang.Object)
//  .tags(java.util.List<CfnTag>)
    .build();
\`\`\`
",
}
`;

exports[`java markdown snapshot 1`] = `
" Initializers <a name=\\"software.amazon.awscdk.services.ecr.CfnPublicRepository.Initializer\\"></a>

\`\`\`java
import software.amazon.awscdk.services.ecr.CfnPublicRepository;

CfnPublicRepository.Builder.create(Construct scope, java.lang.String id)
//  .repositoryCatalogData(java.lang.Object)
//  .repositoryName(java.lang.String)
//  .repositoryPolicyText(java.lang.Object)
//  .tags(java.util.List<CfnTag>)
    .build();
\`\`\`

# \`scope\`<sup>Required</sup> <a name=\\"software.amazon.awscdk.services.ecr.CfnPublicRepository.parameter.scope\\"></a>

- *Type:* [\`software.amazon.awscdk.core.Construct\`](#software.amazon.awscdk.core.Construct)

scope in which this resource is defined.

---

# \`id\`<sup>Required</sup> <a name=\\"software.amazon.awscdk.services.ecr.CfnPublicRepository.parameter.id\\"></a>

- *Type:* \`java.lang.String\`

scoped id of the resource.

---

# \`repositoryCatalogData\`<sup>Optional</sup> <a name=\\"software.amazon.awscdk.services.ecr.CfnPublicRepositoryProps.parameter.repositoryCatalogData\\"></a>

- *Type:* \`java.lang.Object\`

\`AWS::ECR::PublicRepository.RepositoryCatalogData\`.

> [http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-repositorycatalogdata](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-repositorycatalogdata)

---

# \`repositoryName\`<sup>Optional</sup> <a name=\\"software.amazon.awscdk.services.ecr.CfnPublicRepositoryProps.parameter.repositoryName\\"></a>

- *Type:* \`java.lang.String\`

\`AWS::ECR::PublicRepository.RepositoryName\`.

> [http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-repositoryname](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-repositoryname)

---

# \`repositoryPolicyText\`<sup>Optional</sup> <a name=\\"software.amazon.awscdk.services.ecr.CfnPublicRepositoryProps.parameter.repositoryPolicyText\\"></a>

- *Type:* \`java.lang.Object\`

\`AWS::ECR::PublicRepository.RepositoryPolicyText\`.

> [http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-repositorypolicytext](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-repositorypolicytext)

---

# \`tags\`<sup>Optional</sup> <a name=\\"software.amazon.awscdk.services.ecr.CfnPublicRepositoryProps.parameter.tags\\"></a>

- *Type:* java.util.List<[\`software.amazon.awscdk.core.CfnTag\`](#software.amazon.awscdk.core.CfnTag)>

\`AWS::ECR::PublicRepository.Tags\`.

> [http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-tags](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-tags)

---
"
`;

exports[`python json snapshot 1`] = `
Object {
  "id": "aws_cdk.aws_ecr.CfnPublicRepository.Initializer",
  "parameters": Array [
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "- scope in which this resource is defined.",
      "id": "aws_cdk.aws_ecr.CfnPublicRepository.scope",
      "name": "scope",
      "optional": false,
      "type": Object {
        "fqn": "@aws-cdk/core.Construct",
        "name": "aws_cdk.core.Construct",
      },
    },
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "- scoped id of the resource.",
      "id": "aws_cdk.aws_ecr.CfnPublicRepository.id",
      "name": "id",
      "optional": false,
      "type": Object {
        "name": "str",
      },
    },
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "\`AWS::ECR::PublicRepository.RepositoryCatalogData\`.",
      "id": "aws_cdk.aws_ecr.CfnPublicRepositoryProps.repository_catalog_data",
      "name": "repository_catalog_data",
      "optional": true,
      "type": Object {
        "name": "typing.Any",
      },
    },
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "\`AWS::ECR::PublicRepository.RepositoryName\`.",
      "id": "aws_cdk.aws_ecr.CfnPublicRepositoryProps.repository_name",
      "name": "repository_name",
      "optional": true,
      "type": Object {
        "name": "str",
      },
    },
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "\`AWS::ECR::PublicRepository.RepositoryPolicyText\`.",
      "id": "aws_cdk.aws_ecr.CfnPublicRepositoryProps.repository_policy_text",
      "name": "repository_policy_text",
      "optional": true,
      "type": Object {
        "name": "typing.Any",
      },
    },
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "\`AWS::ECR::PublicRepository.Tags\`.",
      "id": "aws_cdk.aws_ecr.CfnPublicRepositoryProps.tags",
      "name": "tags",
      "optional": true,
      "type": Object {
        "name": "typing.List[%]",
        "types": Array [
          Object {
            "fqn": "@aws-cdk/core.CfnTag",
            "name": "aws_cdk.core.CfnTag",
          },
        ],
      },
    },
  ],
  "snippet": "\`\`\`python
import aws_cdk.aws_ecr

aws_cdk.aws_ecr.CfnPublicRepository(
  scope: Construct,
  id: str,
  repository_catalog_data: typing.Any = None,
  repository_name: str = None,
  repository_policy_text: typing.Any = None,
  tags: typing.List[CfnTag] = None
)
\`\`\`
",
}
`;

exports[`python markdown snapshot 1`] = `
" Initializers <a name=\\"aws_cdk.aws_ecr.CfnPublicRepository.Initializer\\"></a>

\`\`\`python
import aws_cdk.aws_ecr

aws_cdk.aws_ecr.CfnPublicRepository(
  scope: Construct,
  id: str,
  repository_catalog_data: typing.Any = None,
  repository_name: str = None,
  repository_policy_text: typing.Any = None,
  tags: typing.List[CfnTag] = None
)
\`\`\`

# \`scope\`<sup>Required</sup> <a name=\\"aws_cdk.aws_ecr.CfnPublicRepository.parameter.scope\\"></a>

- *Type:* [\`aws_cdk.core.Construct\`](#aws_cdk.core.Construct)

scope in which this resource is defined.

---

# \`id\`<sup>Required</sup> <a name=\\"aws_cdk.aws_ecr.CfnPublicRepository.parameter.id\\"></a>

- *Type:* \`str\`

scoped id of the resource.

---

# \`repository_catalog_data\`<sup>Optional</sup> <a name=\\"aws_cdk.aws_ecr.CfnPublicRepositoryProps.parameter.repository_catalog_data\\"></a>

- *Type:* \`typing.Any\`

\`AWS::ECR::PublicRepository.RepositoryCatalogData\`.

> [http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-repositorycatalogdata](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-repositorycatalogdata)

---

# \`repository_name\`<sup>Optional</sup> <a name=\\"aws_cdk.aws_ecr.CfnPublicRepositoryProps.parameter.repository_name\\"></a>

- *Type:* \`str\`

\`AWS::ECR::PublicRepository.RepositoryName\`.

> [http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-repositoryname](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-repositoryname)

---

# \`repository_policy_text\`<sup>Optional</sup> <a name=\\"aws_cdk.aws_ecr.CfnPublicRepositoryProps.parameter.repository_policy_text\\"></a>

- *Type:* \`typing.Any\`

\`AWS::ECR::PublicRepository.RepositoryPolicyText\`.

> [http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-repositorypolicytext](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-repositorypolicytext)

---

# \`tags\`<sup>Optional</sup> <a name=\\"aws_cdk.aws_ecr.CfnPublicRepositoryProps.parameter.tags\\"></a>

- *Type:* typing.List[[\`aws_cdk.core.CfnTag\`](#aws_cdk.core.CfnTag)]

\`AWS::ECR::PublicRepository.Tags\`.

> [http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-tags](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecr-publicrepository.html#cfn-ecr-publicrepository-tags)

---
"
`;

exports[`typescript json snapshot 1`] = `
Object {
  "id": "@aws-cdk/aws-ecr.CfnPublicRepository.Initializer",
  "parameters": Array [
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "- scope in which this resource is defined.",
      "id": "@aws-cdk/aws-ecr.CfnPublicRepository.scope",
      "name": "scope",
      "optional": false,
      "type": Object {
        "fqn": "@aws-cdk/core.Construct",
        "name": "@aws-cdk/core.Construct",
      },
    },
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "- scoped id of the resource.",
      "id": "@aws-cdk/aws-ecr.CfnPublicRepository.id",
      "name": "id",
      "optional": false,
      "type": Object {
        "name": "string",
      },
    },
    Object {
      "default": undefined,
      "deprecated": false,
      "deprecationReason": undefined,
      "docs": "- resource properties.",
      "id": "@aws-cdk/aws-ecr.CfnPublicRepository.props",
      "name": "props",
      "optional": true,
      "type": Object {
        "fqn": "@aws-cdk/aws-ecr.CfnPublicRepositoryProps",
        "name": "@aws-cdk/aws-ecr.CfnPublicRepositoryProps",
      },
    },
  ],
  "snippet": "\`\`\`typescript
import { CfnPublicRepository } from '@aws-cdk/aws-ecr'

new CfnPublicRepository(scope: Construct, id: string, props?: CfnPublicRepositoryProps)
\`\`\`
",
}
`;

exports[`typescript markdown snapshot 1`] = `
" Initializers <a name=\\"@aws-cdk/aws-ecr.CfnPublicRepository.Initializer\\"></a>

\`\`\`typescript
import { CfnPublicRepository } from '@aws-cdk/aws-ecr'

new CfnPublicRepository(scope: Construct, id: string, props?: CfnPublicRepositoryProps)
\`\`\`

# \`scope\`<sup>Required</sup> <a name=\\"@aws-cdk/aws-ecr.CfnPublicRepository.parameter.scope\\"></a>

- *Type:* [\`@aws-cdk/core.Construct\`](#@aws-cdk/core.Construct)

scope in which this resource is defined.

---

# \`id\`<sup>Required</sup> <a name=\\"@aws-cdk/aws-ecr.CfnPublicRepository.parameter.id\\"></a>

- *Type:* \`string\`

scoped id of the resource.

---

# \`props\`<sup>Optional</sup> <a name=\\"@aws-cdk/aws-ecr.CfnPublicRepository.parameter.props\\"></a>

- *Type:* [\`@aws-cdk/aws-ecr.CfnPublicRepositoryProps\`](#@aws-cdk/aws-ecr.CfnPublicRepositoryProps)

resource properties.

---
"
`;
